### PR TITLE
Collapse and Expand all in the scene tree.

### PR DESCRIPTION
**Pull request by MunWolf**
This adds two buttons on the right click menu in the scene tree that allows you to collapse or expand all selected nodes and their children.

image

### CODE REVIEW

**Comment by Chaosus** [comment]
Icons looks not nice, I'd leave these options without icons at all - otherwise you should add it to all other Expand/Collapse places for consistensy

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -2274,6 +2304,15 @@ void SceneTreeDock::_tree_rmb(const Vector2 &p_menu_pos) {
 		}
 	}
 
+	for (int i = 0; i < selection.size(); ++i) {
+		if (selection[i]->get_child_count() > 0) {
+			menu->add_separator();
+			menu->add_icon_shortcut(get_icon("Collapse", "EditorIcons"), ED_GET_SHORTCUT("scene_tree/collapse_all"), TOOL_COLLAPSE_ALL);
+			menu->add_icon_shortcut(get_icon("Expand", "EditorIcons"), ED_GET_SHORTCUT("scene_tree/expand_all"), TOOL_EXPAND_ALL);
```
Comment (Chaosus) [inline editor/scene_tree_dock.cpp]: `Expand` should be first - for conformity

**Comment by Chaosus** [comment]
`scene_tree_dock.cpp` - is not formatted correctly, you should use clang-format on all cpp/h files you have been modified, see - http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

**Comment by Xrayez** [comment]
Already proposed and ready to be merged in #19936. 😕

**Comment by MunWolf** [comment]
Ahh @Xrayez I didn't see your PR Ill update my implementation based on the feedback you got then.  Or if people want yours more I can close this one.

File: editor/scene_tree_dock.cpp [hunk 2]
```diff
@@ -283,6 +287,20 @@ bool SceneTreeDock::_cyclical_dependency_exists(const String &p_target_scene_pat
 	return false;
 }
 
+void _set_fold_recursive(Node *root, TreeItem *p_item, bool p_collapsed) {
+	TreeItem *child = p_item->get_children();
+	if (child != nullptr) {
+		NodePath np = p_item->get_metadata(0);
```
Comment (Xrayez) [inline editor/scene_tree_dock.cpp]: Hmm neat!

**Pull request closed without merging**